app/lib/audit/checksum.rb

```diff
@@ -58,5 +58,20 @@ def self.validate_list_of_druids(druid_list_file_path)
         Checksum.validate_druid(row.first)
       end
     end
+
+    # validate objects with a particular status on a particular endpoint
+    def self.validate_status_root(status, endpoint_name, limit=Settings.c2m_sql_limit)
+      # pres_copies is an AR Relation; it could return a lot of results, so we want to process it in
+      # batches.  we can't use ActiveRecord's .find_each, because that'll disregard the order .fixity_check_expired
+      # specified.  so we use our own batch processing method, which does respect Relation order.
+      pres_copies = PreservedCopy.send(status).by_endpoint_name(endpoint_name).for_online_endpoints
+      desc = "Number of Preserved Copies of status #{status} from #{endpoint_name} to be checksum validated"
+      logger.info "#{desc}: #{pres_copies.count}"
+      ActiveRecordUtils.process_in_batches(pres_copies, limit) do |pc|
+        logger.info "CV beginning for #{pc.preserved_object.druid}; starting status #{pc.status}"
+        ChecksumValidator.new(pc).validate_checksums
```

Comment:
While this is OK, does it make sense to have the PC model provide a method like `validate_checksums!` that wraps a `ChecksumValidator.new(self).validate_checksums`?

Reply:
Not a bad idea, but validating checksums for online pres_copies is diff from archival pres_copies ... seems perhaps better to ticket it than include it here, due to that wrinkle?  (Spoke with @SaravShah who is working on the archival copy C2A and he agrees a follow on ticket is better).  I'll make a ticket.

Reply:
Created #921

Reply:
thx